lib/rubocop/cop/rspec/ordered_hooks.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that before/around/after hooks are defined in the correct order.
+      #
+      # If multiple hooks are defined in example group, they should appear in
+      # the following order:
+      # before :suite
+      # before :context
+      # before :example
+      # around :example
+      # after :example
+      # after :context
+      # after :suite
+      #
+      # @example
+      #  # bad
+      #  after { run_cleanup }
+      #  before { run_setup }
+      #
+      #  # good
+      #  before { run_setup }
+      #  after { run_cleanup }
+      #
+      class OrderedHooks < Base
+        extend AutoCorrector
+
+        MSG = '`%<hook>s` is supposed to appear before `%<previous>s`' \
+          ' at line %<line>d.'
```

Comment:
```suggestion
        MSG = '`%<hook>s` is supposed to appear before `%<previous>s` ' \
          'at line %<line>d.'
```
RuboCop (Parser) unfortunately [cannot check this](https://github.com/rubocop-hq/rubocop/issues/6420).